### ISSUE TITLE
Prevent window from changing the url?

### ISSUE THREAD

**Issue by Cedriking** [nwjs/nw.js#3210]
Hello, I'm updating my app and found something that I dont know how or if its even possible to do?
I have an iframe generated from an async js script, I want to do that when I click inside that iframe the main window (my app) doesnt change his url (location).

How I can do that? if thats not possible, is it possible to prevent the app to change the url anytime? I have only a one-page and dont need my app to change his location at all.

Thank you in advance!

**Comment by mscreenie** [comment]
Have you tried to create the app and test it? nwjs is based on Chromium, you should expect the same behaviour you get in your browser.

If I understand your question you want the node window location to remain the same while you open and navigate in an iframe. I'm doing this and it works fine. However If the child page calls the parent to change its URL it's probably going to change, unless some of node specific attributes prevent this. See https://github.com/nwjs/nw.js/wiki/Security

**Comment by Cedriking** [comment]
Yes I have tried and tested nwjs, I've been using it for quite some time now and everything is great.

And yes that's the issue, in the iframe there is a link, the link i want it to open in openExternal, and prevent the change of the parent (my app) because if not, that replaces my app with a website, 

Thanks a lot for your help

**Comment by mscreenie** [comment]
If you want to open the link using the system's browser, then look at https://github.com/nwjs/nw.js/wiki/App , gui.Shell.openExternal("https://domain.com")

Check the link inside the frame to see if the target is _parent. If this is the case your app URL will change.

In case you don't have access to the Iframe then I'm not sure if you can do much to prevent the Iframe controlling your node app, see the frame security link above, the attributes may help. You may be able to use Javascript to listen for page changes and prevent default.

**Comment by felicienfrancois** [comment]
There are several solutions for your problem.
The easiest one is to intercept links click in your iframe:
```javascript
var iframeElement = document.getElementById("myiframe");
iframeElement.addEventListener("load", function() {
    iframeElement.contentWindow.document.getElementsByTagName("a").forEach(function(link) {
        link.onclick = function(event) {
            event.preventDefault();
            require("nw.gui").Shell.openExternal(this.getAttribute("href"));
            return false;
        };
    });
});
```
With this solution you may have cross domain policy issues depending on your setup

Other solutions I have in mind are
- use [inject-js-start or inject-js-end](https://github.com/nwjs/nw.js/wiki/Manifest-format#inject-js-start--inject-js-end) to inject a javascript code that do the same
- use [chrome webrequest API](https://developer.chrome.com/extensions/webRequest#type-RequestFilter) to intercept and cancel navigation. This API is newly available in NWJS 0.12.0 but is not enough documented by chrome to be trivial enough to use.

**Comment by Cedriking** [comment]
@felicienfrancois :+1: Thank's a lot! That was exactly what I was looking for, I didnt knew that I could add an eventListener to something inside an iframe wow :smile:
